lib/puppet/resource_api/parameter.rb

```diff
@@ -13,7 +13,7 @@ class Puppet::ResourceApi::Parameter < Puppet::Parameter
   # @param attribute_name the name of attribue of the parameter
   # @param resource_hash the resource hash instance which is passed to the
   # parent class.
-  def initialize(type_name, data_type, attribute_name, resource_hash)
+  def initialize(type_name, data_type, attribute_name, resource_hash, _referrable_type)
```

Comment:
Is `Puppet::ResourceApi::Parameter.new` public? If so, then this is a breaking change and I think you'll want to provide a default value for `_referrable_type`.

Reply:
```suggestion
  def initialize(type_name, data_type, attribute_name, resource_hash, _referrable_type = nil)
```

I didn't consider it to be public, but you're right, it is good practice to treat it like that.